ast2vec/cloning.py

```diff
@@ -0,0 +1,213 @@
+            return
+
+        env = os.environ.copy()
+        env["GIT_TERMINAL_PROMPT"] = "0"
+        self._log.info("Cloning from %s...", git_url)
+
+        try:
+            subprocess.check_output(["git", "clone", "--depth=1", git_url, repo_dir],
+                                    env=env, stdin=subprocess.DEVNULL, stderr=subprocess.STDOUT)
+        except subprocess.CalledProcessError as e:
+            shutil.rmtree(repo_dir, ignore_errors=True)
+            self._log.error("Git failed to clone repo. git stderr:\n\t" +
+                            "\n\t".join(e.output.decode("utf8").split("\n")))
+            if not ignore:
+                raise e from None
+            else:
+                return
+        except Exception as e:
+            shutil.rmtree(repo_dir, ignore_errors=True)
+            self._log.error("Unknown error in RepoCloner. Git failed to clone repo.")
+            raise e from None
+
+        self._log.info("Finished cloning %s", git_url)
+        return repo_dir
+
+    def classify_repo(self, repo_dir: str) -> dict:
+        """
+        Classify files in a repository using enry.
+
+        :param repo_dir: Path to repository directory.
+        :return: enry output loaded from JSON.
+        """
+        if not self._linguist:
+            raise LinguistFailedError("Linguist is not set - cannot classify the files")
+
+        self._log.info("Classifying the files...")
+        repo_dir = os.path.abspath(repo_dir)
+        cmdline = [self._linguist]
+        if self._is_enry:
+            cmdline += ["-json", repo_dir]
+        else:
+            cmdline += [repo_dir, "--json"]
+        try:
+            bjson = subprocess.check_output(cmdline)
+        except subprocess.CalledProcessError:
+            self._log.error("Couldn't classify files in %s", repo_dir)
+            raise LinguistFailedError() from None
+        classified = json.loads(bjson.decode("utf8"))
+        self._log.info("Result: %s", {k: len(v) for k, v in classified.items()})
+        return classified
+
+    def cleanup_repo(self, classified: dict, repo_dir: str) -> None:
+        """
+        Delete files not classified by enry (if some languages were specified) then only them
+        are preserved.
+
+        :param classified: enry output loaded from JSON.
+        :param repo_dir: Path to repository directory.
+        :return:
+        """
+        if self._languages:
+            self._log.info("Removing files with languages not in (%s)",
+                           ",".join(self._languages))
+            languages = self._languages
+        else:
+            self._log.info("Removing files not classified by enry.")
+            languages = classified.keys()
+
+        allowed_files = set(str.encode(os.path.join(repo_dir, fname)) for lang in languages
+                            if lang in classified for fname in classified[lang])
+
+        for dirname, dirnames, filenames in os.walk(str.encode(repo_dir)):
```

Comment:
I feel that it's better to be explicit in here – in case someone wants to modify this part, or people might want to run this code in their Jupyter Notebook, which will slightly mess with the value of `_`.
Anyways, I'll probably use `dirnames` to remove empty directories as you suggested 😄